config/webpack.config.prod.js

```diff
@@ -36,26 +37,72 @@ if (env.stringified['process.env'].NODE_ENV !== '"production"') {
 
-// ExtractTextPlugin expects the build output to be flat.
-// (See https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/27)
-// However, our output is structured with css, js and media folders.
-// To have this structure working with relative paths, we have to use custom options.
-const extractTextPluginOptions = shouldUseRelativeAssetPaths
+// ExtractTextPlugin required its options.publicPath to be overloaded so that relative paths work
+//    ExtractTextPlugin expects the build output to be flat.
+//    (See https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/27)
+//    However, our output is structured with css, js and media folders.
+//    To have this structure working with relative paths, we have to use custom options.
+// Now that we replaced ExtractTextPlugin with MiniCssExtractPlugin, I don't think it is necessary
+// Keeping it here just in case though.
+const miniCssExtractPluginOptions = shouldUseRelativeAssetPaths
   ? // Making sure that the publicPath goes back to to build folder.
-    { publicPath: Array(cssFilename.split('/').length).join('../') }
+  { publicPath: Array(cssFilename.split('/').length).join('../') }
   : {};
 
 // This is the production configuration.
 // It compiles slowly and is focused on producing a fast and minimal bundle.
 // The development configuration is different and lives in a separate file.
 module.exports = {
+  mode: 'production',
   // Don't attempt to continue if there are any errors.
   bail: true,
   // We generate sourcemaps in production. This is slow but gives good results.
   // You can exclude the *.map files from the build during deployment.
   devtool: shouldUseSourceMap ? 'source-map' : false,
   entry: {
-    main: [require.resolve('./polyfills'), paths.appIndexJs]
+    main: [require.resolve('./polyfills'), paths.appIndexJs],
+  },
+  // Terser Plugin helps reserve keywords such as ECPair (used for logging into bitgo)
+  // TerserPlugin is automatically added in Webpack 5, but not Webpack 4,
+  optimization: {
+    minimize: true,
+    minimizer: [
+      new TerserPlugin({
+        terserOptions: {
+          warnings: false,
+          compress: {
+            comparisons: false,
+          },
+          parse: {},
+          mangle: {
+            reserved: ['BigInteger', 'ECPair', 'Point']
+          },
+          output: {
+            comments: false,
+            ascii_only: true,
+          },
+        },
+        parallel: true,
+      }),
+    ],
+    nodeEnv: 'production',
+    sideEffects: true,
+    concatenateModules: true,
+    runtimeChunk: 'single',
+    splitChunks: {
+      chunks: 'all',
+      maxInitialRequests: 10,
+      minSize: 0,
+      cacheGroups: {
+        vendor: {
+          test: /[\\/]node_modules[\\/]/,
+          name(module) {
+            const packageName = module.context.match(/[\\/]node_modules[\\/](.*?)([\\/]|$)/)[1];
+            return `npm.${packageName.replace('@', '')}`;
```

Comment:
@bjon-bitgo can you turn this into a function and write some tests for it

Reply:
will be a lot of work. I dont think WRW has unit tests even set up.